src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -16,25 +12,34 @@
 using System.Threading;
 using System.Threading.Tasks;
 using System.Xml;
-#if MANAGED_SNI
-using System.Data.SqlClient.SNI;
-#endif
+
 
 using MSS = Microsoft.SqlServer.Server;
 
 namespace System.Data.SqlClient
 {
+
+    struct SNIErrorDetails
+    {
+        public string errorMessage;
+        public uint nativeError;
+        public uint sniErrorNumber;
+        public int provider;
+        public uint lineNumber;
+        public string function;
+        public Exception exception;
+    }
+
     // The TdsParser Object controls reading/writing to the netlib, parsing the tds,
     // and surfacing objects to the user.
-    sealed internal class TdsParser
+    sealed internal partial class TdsParser
     {
         static TdsParser()
         {
             // For CoreCLR, we need to register the ANSI Code Page encoding provider before attempting to get an Encoding from a CodePage
             Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```

Comment:
@tarekgh SqlClient needs to provide the encoding related abstractions to the consumers of Sql client based on the exchange with SqlServer. 
I don't think it makes sense for SqlClient to expect consumers to set the encoding provider. 
Am I missing something here? Regardless if this needs to be addresses I will open another issue for this item. 
Let me know if my understanding is incorrect.